management/src/main/java/org/ehcache/management/registry/DefaultCollectorService.java

```diff
@@ -113,6 +121,20 @@ public void cacheRemoved(String alias, Cache<?, ?> cache) {
         EhcacheNotification.CACHE_REMOVED.name()));
   }
 
+  private void cacheCleared(String alias) {
+    collector.onNotification(
+      new ContextualNotification(
+        configuration.getContext().with("cacheName", alias),
+        EhcacheNotification.CACHE_CLEARED.name()));
+  }
+
+  private void registerClearNotification(String alias, Cache<?, ?> cache) {
+    OperationStatistic<ClearOutcome> clear = StatsUtils.findOperationStatisticOnChildren(cache,
+      ClearOutcome.class, "clear");
+    clear.addDerivedStatistic(new OperationResultFilter<>(EnumSet.of(ClearOutcome.SUCCESS),
```

Comment:
This looks like a hack to me... Using the statistics to get the information about a clear. I don't like it at all. The statistics should not carry business operations. I would prefer the listener to be called by `EhcacheBase.clear()`. But that would mean having it available in `EhcacheBase`. Can't you work with the `CacheEventDispatcher`?

Reply:
hello @henri-tremblay can you please provide guidance on how to this differently ? There is no ```cacheCleared``` in  ```CacheManagerListener```
How could we implement ```cacheCleared``` in ```CacheEventDispatcher``` ?
The "hack" suggested in a 3 lines snippet is totally limited in scope, and actually, pretty readable and understandable, and will provide an important feature for users. If you think there's a "cleaner" way to implement this feature, please share it with us

Reply:
This looks like a hack indeed, but a reasonably robust one since it relies on the already existing cache clear statistic. It's also fairly elegant since the hackery is rather self-contained.

Firing this notification from `EhcacheBase` would take a much larger effort which probably isn't worth it since we do not want to expose cache clearing events through the API as the only use-case that makes sense for this notification is M&M.

Reply:
Ok. Going with the family then.